extensions/src/main/java/org/pentaho/platform/web/http/api/resources/services/FileService.java

```diff
@@ -289,6 +289,15 @@ public List<Setting> doGetPathsAccessList( StringListWrapper pathsWrapper ) {
     return pathsPermissonsSettings;
   }
 
+  public void doesFileExists( String destPathId ) throws Exception {
+    String idToPath = idToPath( destPathId );
```

Comment:
Can we just call `JcrRepositoryFileDao.java#getFile` and check the return value and/or catch exceptions instead of creating adding an interface function to IUnifiedRepositoryWebService.java?

If there are other use cases or code, then I can see adding `doesFileExist() `to all the code layers related to  IUnifiedRepositoryWebService.java

Reply:
Hi @NJtwentyone,
Thanks for reviewing and sharing your thoughts.

Requirement - Before saving a file, whether there is a file available already should be checked. Went thru the existing code in platform and scheduler-plugin, only found `getFile` But in `getFile` Stream is created and content of the file is getting loaded.
As per the requirement, just a check would suffice whether a file exists or not. do not need to load the content of the file. And that is why `doesFileExist` had to be added. Could not find any other alternative.

If you have any suggestions around or references available whether we can check in any other efficient ways please do share.

Reply:
Hi @NJtwentyone Created this new PR addressing this https://github.com/pentaho/pentaho-platform/pull/5815

Reply:
https://github.com/pentaho/pentaho-platform/pull/5815 seem alot simpler and still allows still allows for modular unit testing.

Reply:
@NJtwentyone Before closing this PR would like to confirm once
https://github.com/pentaho/pentaho-platform/blob/master/repository/src/main/java/org/pentaho/platform/repository2/unified/jcr/JcrRepositoryFileDao.java#L892-L897 
this code is checking for file in a tenant eg., /pentaho/tenant0/home/admin/testfile.ktr

And getFile only checks for /home/admin/testfile.ktr

Hope PR #5815 would not fail in a multi-tenant environment.

Reply:
off hand, I don't know the internals of JCR code that low level. Looking at JCR filesystem through database queries and/or code is difficult. 

If you do black box testing, you should either get a file back with the data or not.

 
The JCR code is very hard to read, hard to debug and most of that functionality has been tested over several releases.

My overall sentiment is, DRY - "Don't Repeat Yourself"

There already is a collection of interfaces/services/dao's that already return some type of file object when given some arguments. The light weight option, is to just call the previously implemented function(s) and then return that value and/or catch an exception.